FirebaseAdmin/FirebaseAdmin/Auth/Hash/Hmac.cs

```diff
@@ -0,0 +1,58 @@
+// Copyright 2020, Google Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace FirebaseAdmin.Auth.Hash
+{
+    /// <summary>
+    /// Base class for Hmac type hashes.
+    /// </summary>
+    public abstract class Hmac : UserImportHash
```

Comment:
Can this be internal or protected?

Reply:
Based on my attempt yesterday it looks like no, but the constructor can remain internal. This is due to "Inconsistent accessibility"